File: linear-intellij-plugin/src/main/java/com/linear/intellij/toolwindow/LinearToolWindowPanel.java
package com.linear.intellij.toolwindow;

import com.intellij.openapi.actionSystem.ActionManager;
import com.intellij.openapi.actionSystem.ActionToolbar;
import com.intellij.openapi.actionSystem.DefaultActionGroup;
import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.SimpleToolWindowPanel;
import com.intellij.ui.JBSplitter;
import com.intellij.ui.ScrollPaneFactory;
import com.intellij.ui.components.JBLabel;
import com.intellij.ui.components.JBList;
import com.intellij.ui.components.JBScrollPane;
import com.intellij.util.ui.JBUI;
import com.linear.intellij.models.LinearTask;
import com.linear.intellij.services.LinearApiService;
import com.linear.intellij.ui.TaskDetailsPanel;
import com.linear.intellij.ui.TaskListCellRenderer;

import javax.swing.*;
import java.awt.*;
import java.util.List;

public class LinearToolWindowPanel extends SimpleToolWindowPanel {
    private final Project project;
    private final JBList<LinearTask> taskList;
    private final DefaultListModel<LinearTask> listModel;
    private final TaskDetailsPanel detailsPanel;
    private final JPanel mainPanel;
    
    public LinearToolWindowPanel(Project project) {
        super(true, true);
        this.project = project;
        this.listModel = new DefaultListModel<>();
        this.taskList = new JBList<>(listModel);
        this.detailsPanel = new TaskDetailsPanel();
        this.mainPanel = new JPanel(new BorderLayout());
        
        initComponents();
        loadTasks();
    }
    
    private void initComponents() {
        taskList.setCellRenderer(new TaskListCellRenderer());
        taskList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
        taskList.addListSelectionListener(e -> {
            if (!e.getValueIsAdjusting()) {
                LinearTask selected = taskList.getSelectedValue();
                detailsPanel.setTask(selected);
            }
        });
        
        JBScrollPane scrollPane = new JBScrollPane(taskList);
        scrollPane.setBorder(JBUI.Borders.empty());
        
        JBSplitter splitter = new JBSplitter(false, 0.6f);
        splitter.setFirstComponent(scrollPane);
        splitter.setSecondComponent(detailsPanel);
        
        mainPanel.add(splitter, BorderLayout.CENTER);
        
        setContent(mainPanel);
        setupToolbar();
    }
    
    private void setupToolbar() {
        DefaultActionGroup actionGroup = new DefaultActionGroup();
        actionGroup.add(ActionManager.getInstance().getAction("com.linear.intellij.actions.RefreshLinearTasks"));
        
        ActionToolbar toolbar = ActionManager.getInstance()
                .createActionToolbar("LinearToolbar", actionGroup, true);
        toolbar.setTargetComponent(mainPanel);
        setToolbar(toolbar.getComponent());
    }
    
    public void loadTasks() {
        LinearApiService apiService = ApplicationManager.getApplication()
                .getService(LinearApiService.class);
        
        if (!apiService.isConfigured()) {
            showConfigurationMessage();
            return;
        }
        
        SwingUtilities.invokeLater(() -> {
            listModel.clear();
            mainPanel.removeAll();
            mainPanel.add(new JBLabel("Loading tasks...", SwingConstants.CENTER), BorderLayout.CENTER);
            mainPanel.revalidate();
            mainPanel.repaint();
        });
        
        apiService.fetchAssignedTasks()
                .thenAccept(this::updateTaskList)
                .exceptionally(throwable -> {
                    SwingUtilities.invokeLater(() -> showErrorMessage(throwable.getMessage()));
                    return null;
                });
    }
    
    private void updateTaskList(List<LinearTask> tasks) {
        SwingUtilities.invokeLater(() -> {
            listModel.clear();
            tasks.forEach(listModel::addElement);
            
            mainPanel.removeAll();
            if (tasks.isEmpty()) {
                mainPanel.add(new JBLabel("No tasks found", SwingConstants.CENTER), BorderLayout.CENTER);
            } else {
                initComponents();
            }
            mainPanel.revalidate();
            mainPanel.repaint();
        });
    }
    
    private void showConfigurationMessage() {
        mainPanel.removeAll();
        JPanel messagePanel = new JPanel(new BorderLayout());
        JBLabel message = new JBLabel("<html><center>Linear API key not configured.<br>Please configure it in Settings > Tools > Linear</center></html>", 
                SwingConstants.CENTER);
        messagePanel.add(message, BorderLayout.CENTER);
        mainPanel.add(messagePanel, BorderLayout.CENTER);
        mainPanel.revalidate();
        mainPanel.repaint();
    }
    
    private void showErrorMessage(String error) {
        mainPanel.removeAll();
        mainPanel.add(new JBLabel("Error: " + error, SwingConstants.CENTER), BorderLayout.CENTER);
        mainPanel.revalidate();
        mainPanel.repaint();
    }
}

File: src/main/kotlin/com/linear/intellij/toolwindow/LinearToolWindowPanel.kt
package com.linear.intellij.toolwindow

import com.intellij.openapi.actionSystem.ActionManager
import com.intellij.openapi.actionSystem.DefaultActionGroup
import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.SimpleToolWindowPanel
import com.intellij.ui.JBSplitter
import com.intellij.ui.components.JBLabel
import com.intellij.ui.components.JBList
import com.intellij.ui.components.JBScrollPane
import com.intellij.util.ui.JBUI
import com.linear.intellij.models.LinearTask
import com.linear.intellij.services.LinearApiService
import com.linear.intellij.ui.TaskDetailsPanel
import com.linear.intellij.ui.TaskListCellRenderer
import java.awt.BorderLayout
import javax.swing.*

class LinearToolWindowPanel(private val project: Project) : SimpleToolWindowPanel(true, true) {
    private val listModel = DefaultListModel<LinearTask>()
    private val taskList = JBList(listModel)
    private val detailsPanel = TaskDetailsPanel()
    private val mainPanel = JPanel(BorderLayout())
    
    init {
        initComponents()
        loadTasks()
    }
    
    private fun initComponents() {
        taskList.cellRenderer = TaskListCellRenderer()
        taskList.selectionMode = ListSelectionModel.SINGLE_SELECTION
        taskList.addListSelectionListener { e ->
            if (!e.valueIsAdjusting) {
                val selected = taskList.selectedValue
                detailsPanel.setTask(selected)
            }
        }
        
        val scrollPane = JBScrollPane(taskList).apply {
            border = JBUI.Borders.empty()
        }
        
        val splitter = JBSplitter(false, 0.6f).apply {
            firstComponent = scrollPane
            secondComponent = detailsPanel
        }
        
        mainPanel.add(splitter, BorderLayout.CENTER)
        
        setContent(mainPanel)
        setupToolbar()
    }
    
    private fun setupToolbar() {
        val actionGroup = DefaultActionGroup()
        actionGroup.add(ActionManager.getInstance().getAction("com.linear.intellij.actions.RefreshLinearTasks"))
        
        val toolbar = ActionManager.getInstance()
            .createActionToolbar("LinearToolbar", actionGroup, true)
        toolbar.setTargetComponent(mainPanel)
        setToolbar(toolbar.component)
    }
    
    fun loadTasks() {
        val apiService = ApplicationManager.getApplication()
            .getService(LinearApiService::class.java)
        
        if (!apiService.isConfigured()) {
            showConfigurationMessage()
            return
        }
        
        SwingUtilities.invokeLater {
            listModel.clear()
            mainPanel.removeAll()
            mainPanel.add(JBLabel("Loading tasks...", SwingConstants.CENTER), BorderLayout.CENTER)
            mainPanel.revalidate()
            mainPanel.repaint()
        }
        
        apiService.fetchAssignedTasks()
            .thenAccept(::updateTaskList)
            .exceptionally { throwable ->
                SwingUtilities.invokeLater { showErrorMessage(throwable.message ?: "Unknown error") }
                null
            }
    }
    
    private fun updateTaskList(tasks: List<LinearTask>) {
        SwingUtilities.invokeLater {
            listModel.clear()
            tasks.forEach(listModel::addElement)
            
            mainPanel.removeAll()
            if (tasks.isEmpty()) {
                mainPanel.add(JBLabel("No tasks found", SwingConstants.CENTER), BorderLayout.CENTER)
            } else {
                initComponents()
            }
            mainPanel.revalidate()
            mainPanel.repaint()
        }
    }
    
    private fun showConfigurationMessage() {
        mainPanel.removeAll()
        val messagePanel = JPanel(BorderLayout())
        val message = JBLabel(
            "<html><center>Linear API key not configured.<br>Please configure it in Settings > Tools > Linear</center></html>",
            SwingConstants.CENTER
        )
        messagePanel.add(message, BorderLayout.CENTER)
        mainPanel.add(messagePanel, BorderLayout.CENTER)
        mainPanel.revalidate()
        mainPanel.repaint()
    }
    
    private fun showErrorMessage(error: String) {
        mainPanel.removeAll()
        mainPanel.add(JBLabel("Error: $error", SwingConstants.CENTER), BorderLayout.CENTER)
        mainPanel.revalidate()
        mainPanel.repaint()
    }
}